scripts/obtain/install.ps1

```diff
@@ -3,106 +3,219 @@
+function Get-User-Share-Path() {
+    Say-Invocation $MyInvocation
+
+    $InstallRoot = $env:DOTNET_INSTALL_DIR
+    if (!$InstallRoot) {
+        $InstallRoot = "$env:LocalAppData\Microsoft\.dotnet"
+    }
+    return $InstallRoot
+}
+
+function Resolve-Installation-Path([string]$InstallDir) {
+    Say-Invocation $MyInvocation
+
+    if ($InstallDir -eq "<usershare>") {
+        return Get-User-Share-Path
+    }
+    return $InstallDir
+}
+
+# TODO: check for global.json
+function Get-Installed-Version-Info([string]$InstallRoot) {
+    Say-Invocation $MyInvocation
+
+    $VersionFile = $InstallRoot + $LocalVersionFileRelativePath
+    Say-Verbose "Local version file: $VersionFile"
+    
+    if (Test-Path $VersionFile) {
+        $VersionText = cat $VersionFile
+        Say-Verbose "Local version file text: $VersionText"
+        return Get-Version-Info-From-Version-Text $VersionText
+    }
+
+    Say-Verbose "Local version file not found."
+
+    return $null
+}
 
-say "Downloading $DotNetFileName from $DotNetUrl"
-$resp = Invoke-WebRequest -UseBasicParsing "$DotNetUrl/$DotNetFileName" -OutFile "$InstallDir\$DotNetFileName"
+function Get-Absolute-Path([string]$RelativeOrAbsolutePath) {
+    # Too much spam
+    # Say-Invocation $MyInvocation
 
-say "Extracting zip"
+    return $ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPath($RelativeOrAbsolutePath)
+}
 
-# Create the destination
-if (Test-Path "$InstallDir\cli_new") {
-    del -rec -for "$InstallDir\cli_new"
+function Extract-And-Override-Zip([string]$ZipPath, [string]$OutPath) {
+    Say-Invocation $MyInvocation
+
+    Add-Type -Assembly System.IO.Compression.FileSystem | Out-Null
+    Set-Variable -Name Zip
+    try {
+        $Zip = [System.IO.Compression.ZipFile]::OpenRead($ZipPath)
+        
+        foreach ($entry in $Zip.Entries) {
+            $DestinationPath = Get-Absolute-Path $(Join-Path -Path $OutPath -ChildPath $entry.FullName)
+            $DestinationDir = Split-Path -Parent $DestinationPath
+            New-Item -ItemType Directory -Force -Path $DestinationDir | Out-Null
+            [System.IO.Compression.ZipFileExtensions]::ExtractToFile($entry, $DestinationPath, $true)
+        }
+    }
+    finally {
+        if ($Zip -ne $null) {
+            $Zip.Dispose()
+        }
+    }
 }
-mkdir "$InstallDir\cli_new" | Out-Null
 
-Add-Type -Assembly System.IO.Compression.FileSystem | Out-Null
-[System.IO.Compression.ZipFile]::ExtractToDirectory("$InstallDir\$DotNetFileName", "$InstallDir\cli_new")
+$AzureChannel = Get-Azure-Channel-From-Channel -Channel $Channel
+$CLIArchitecture = Get-CLIArchitecture-From-Architecture $Architecture
+$SpecificVersion = Get-Specific-Version-From-Version -AzureFeed $AzureFeed -AzureChannel $AzureChannel -CLIArchitecture $CLIArchitecture -Version $Version
+$DownloadLink = Construct-Download-Link -AzureFeed $AzureFeed -AzureChannel $AzureChannel -SpecificVersion $SpecificVersion -CLIArchitecture $CLIArchitecture
```

Comment:
Is possible like in here - https://github.com/dotnet/cli/blob/rel/1.0.0/packaging/windows/generatemsi.ps1#L68.